DicomRtImportExport/Logic/vtkSlicerDicomRtImportExportModuleLogic.cxx

```diff
@@ -805,243 +803,622 @@ bool vtkSlicerDicomRtImportExportModuleLogic::vtkInternal::LoadExternalBeamPlan(
+  beamModelHierarchyDisplayNode->SetVisibility(1);
+  scene->AddNode(beamModelHierarchyDisplayNode);
+  beamModelHierarchyNode->SetAndObserveDisplayNodeID( beamModelHierarchyDisplayNode->GetID() );
+
+  // Hide beam polydata
+  vtkMRMLModelDisplayNode* displayNode = vtkMRMLModelDisplayNode::SafeDownCast(beamNode->GetDisplayNode());
+  if (displayNode)
+  {
+    displayNode->VisibilityOff();
+  }
+
+  // Set beam node as a parent for observed nodes
+  vtkIdType beamShId = vtkMRMLSubjectHierarchyNode::INVALID_ITEM_ID;
+  vtkIdType mlcTableShId = vtkMRMLSubjectHierarchyNode::INVALID_ITEM_ID;
+  // get beam subject hierarchy ID
+  beamShId = shNode->GetItemByDataNode(beamNode);
+  // put observed mlc data under beam node parent
+  if (mlcTableNode)
+  {
+    mlcTableShId = shNode->GetItemByDataNode(mlcTableNode);
+  }
+  if (beamShId != vtkMRMLSubjectHierarchyNode::INVALID_ITEM_ID && 
+    mlcTableShId != vtkMRMLSubjectHierarchyNode::INVALID_ITEM_ID)
+  {
+    shNode->SetItemParent( mlcTableShId, beamShId);
+  }
+
+  // put observed scan spot data under ion beam node parent
+  if (ionBeamNode)
+  {
+    vtkIdType scanSpotShId = vtkMRMLSubjectHierarchyNode::INVALID_ITEM_ID;
+    if (scanSpotTableNode)
+    {
+      scanSpotShId = shNode->GetItemByDataNode(scanSpotTableNode);
+    }
+
+    if (beamShId != vtkMRMLSubjectHierarchyNode::INVALID_ITEM_ID && 
+      scanSpotShId != vtkMRMLSubjectHierarchyNode::INVALID_ITEM_ID)
+    {
+      shNode->SetItemParent( scanSpotShId, beamShId);
+    }
+  }
+
+  return beamNode;
+}
+
+//---------------------------------------------------------------------------
+bool vtkSlicerDicomRtImportExportModuleLogic::vtkInternal::LoadDynamicBeamSequence( 
+  vtkSlicerDicomRtReader* rtReader, const char* seriesName, 
+  vtkMRMLRTPlanNode* planNode, int beamIndex, vtkMRMLRTBeamNode* proxyBeamNode, 
+  vtkMRMLLinearTransformNode* proxyTransformNode, 
+  vtkMRMLTableNode* proxyMlcTableNode, vtkMRMLTableNode* proxyScanSpotTableNode)
+{
+  vtkMRMLScene* scene = planNode->GetScene();
+  this->External->BeamsLogic->SetMRMLScene(scene);
+
+  vtkMRMLSubjectHierarchyNode* shNode = vtkMRMLSubjectHierarchyNode::GetSubjectHierarchyNode(scene);
+  if (!shNode)
+  {
+    vtkErrorWithObjectMacro(this->External, "LoadDynamicBeamSequence: Failed to access subject hierarchy node");
+    return false;
+  }
+  vtkSmartPointer<vtkMRMLModelHierarchyNode> beamModelHierarchyRootNode;
+
+  unsigned int dicomBeamNumber = rtReader->GetBeamNumberForIndex(beamIndex);
+  const char* beamName = rtReader->GetBeamName(dicomBeamNumber);
+  unsigned int nofControlPoints = rtReader->GetBeamNumberOfControlPoints(dicomBeamNumber);
+  const char* beamType = rtReader->GetBeamType(dicomBeamNumber);
+  const char* treatmentDeliveryType = rtReader->GetBeamTreatmentDeliveryType(dicomBeamNumber);
+
+  // Create sequence node for RTBeam, transformation, table
+  auto beamSequenceNode = vtkSmartPointer<vtkMRMLSequenceNode>::New();

```

Comment:
vtkNew<vtkMRMLSequenceNode> beamSequenceNode; is a bit simpler syntax (3 words instead of 5)